core/metamodel.class.php

```diff
@@ -4219,40 +4219,75 @@ public static function AddMagicPlaceholders($aPlaceholders, $aExpectedArgs = nul
 		}
 		else
 		{
-			$aCurrentUser = array();
-			$aCurrentContact = array();
+			$aCurrentUser = [];
+			$aCurrentContact = [];
 			foreach ($aExpectedArgs as $expression)
 			{
 				$aName = explode('->', $expression->GetName());
 				if ($aName[0] == 'current_contact_id') {
 					$aPlaceholders['current_contact_id'] = UserRights::GetContactId();
-				}
-				if ($aName[0] == 'current_user') {
+				} else if ($aName[0] == 'current_user') {
 					array_push($aCurrentUser, $aName[1]);
-				}
-				if ($aName[0] == 'current_contact') {
+				} else if ($aName[0] == 'current_contact') {
 					array_push($aCurrentContact, $aName[1]);
 				}
 			}
 			if (count($aCurrentUser) > 0) {
-				$oUser = UserRights::GetUserObject();
-				$aPlaceholders['current_user->object()'] = $oUser;
-				foreach ($aCurrentUser as $sField) {
-					$aPlaceholders['current_user->'.$sField] = $oUser->Get($sField);
-				}
+				static::FillPlaceholders($aPlaceholders, 'current_user', UserRights::GetUserObject(), $aCurrentUser);
 			}
 			if (count($aCurrentContact) > 0) {
-				$oPerson = UserRights::GetContactObject();
-				$aPlaceholders['current_contact->object()'] = $oPerson;
-				foreach ($aCurrentContact as $sField) {
-					$aPlaceholders['current_contact->'.$sField] = $oPerson->Get($sField);
-				}
+				static::FillPlaceholders($aPlaceholders, 'current_contact', UserRights::GetContactObject(), $aCurrentContact);
 			}
 		}
 
 		return $aPlaceholders;
 	}
 
+	/**
+	 * @since 3.1 N°6824
+	 * @param array $aPlaceholders
+	 * @param string $sPlaceHolderPrefix
+	 * @param \DBObject|null $oObject
+	 * @param array $aCurrentUser
+	 *
+	 * @return void
+	 *
+	 */
+	private static function FillPlaceholders(array &$aPlaceholders, string $sPlaceHolderPrefix, $oObject, $aCurrentUser) {
```

Comment:
```suggestion
	 * @param array $aPlaceholders
	 * @param string $sPlaceHolderPrefix
	 * @param ?\DBObject $oObject
	 * @param array $aCurrentUser
	 *
	 * @return void
	 *
	 */
	private static function FillPlaceholders(array &$aPlaceholders, string $sPlaceHolderPrefix, ?\DBObject $oObject, array $aCurrentUser) {
```

[Supported since php 7.1+](https://3v4l.org/SSLRN)

Reply:
@Hipska Good idea, can you add the return type hint to your suggestion as well?
@odain-cbd How about renaming the method `FillObjectPlaceholders`? As it concern only placeholders with "->object()", we might introduce a similar `FillPlaceholders` method for scalars in the futur.